File: oposapp/lib/models/pregunta.dart
class Pregunta {
  final int id;
  final int testId;
  final String textoPregunta;
  final String opcionA;
  final String opcionB;
  final String opcionC;
  final String opcionD;
  final String respuestaCorrecta; // 'A', 'B', 'C', 'D'
  final String explicacion;
  final String tema;
  final String oposicion;
  final String dificultad;

  Pregunta({
    required this.id,
    required this.testId,
    required this.textoPregunta,
    required this.opcionA,
    required this.opcionB,
    required this.opcionC,
    required this.opcionD,
    required this.respuestaCorrecta,
    required this.explicacion,
    required this.tema,
    required this.oposicion,
    required this.dificultad,
  });

  factory Pregunta.fromJson(Map<String, dynamic> json) {
    return Pregunta(
      id: json['id'] ?? 0,
      testId: json['testId'] ?? 0,
      textoPregunta: json['textoPregunta'] ?? '',
      opcionA: json['opcionA'] ?? '',
      opcionB: json['opcionB'] ?? '',
      opcionC: json['opcionC'] ?? '',
      opcionD: json['opcionD'] ?? '',
      respuestaCorrecta: json['respuestaCorrecta'] ?? 'A',
      explicacion: json['explicacion'] ?? '',
      tema: json['tema'] ?? '',
      oposicion: json['oposicion'] ?? '',
      dificultad: json['dificultad'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'testId': testId,
      'textoPregunta': textoPregunta,
      'opcionA': opcionA,
      'opcionB': opcionB,
      'opcionC': opcionC,
      'opcionD': opcionD,
      'respuestaCorrecta': respuestaCorrecta,
      'explicacion': explicacion,
      'tema': tema,
      'oposicion': oposicion,
      'dificultad': dificultad,
    };
  }
}

File: api-backend/src/main/java/es/ivanesco/oposapp/api/models/Pregunta.java
package es.ivanesco.oposapp.api.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "preguntas", schema = "tfg")
public class Pregunta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "test_id")
    private Integer testId;

    @Column(name = "texto_pregunta", nullable = false, columnDefinition = "TEXT")
    private String textoPregunta;

    @Column(name = "opcion_a", nullable = false, columnDefinition = "TEXT")
    private String opcionA;

    @Column(name = "opcion_b", nullable = false, columnDefinition = "TEXT")
    private String opcionB;

    @Column(name = "opcion_c", nullable = false, columnDefinition = "TEXT")
    private String opcionC;

    @Column(name = "opcion_d", nullable = false, columnDefinition = "TEXT")
    private String opcionD;

    @Column(name = "respuesta_correcta", length = 1, columnDefinition = "bpchar")
    private String respuestaCorrecta;

    @Column(columnDefinition = "TEXT")
    private String explicacion;

    @Column(length = 300)
    private String tema;

    @Column(length = 200)
    private String oposicion;

    @Column(length = 20)
    private String dificultad;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "solicitud_id")
    @JsonIgnore  // No serializar — evita LazyInitializationException
    private SolicitudGeneracion solicitud;

    @CreationTimestamp
    @Column(name = "fecha_creacion", updatable = false)
    private LocalDateTime fechaCreacion;
}
